### PR TITLE
Update to JSX3

### PR DESCRIPTION
**Pull request by Anahkiasen**
This makes the components usable in a codebase using the [v3 implementation of JSX](https://reasonml.github.io/reason-react/blog/2019/04/10/react-hooks) following the new [documentation](https://reasonml.github.io/reason-react/docs/en/components#interop)

### CODE REVIEW

**Comment by Anahkiasen** [comment]
Seems I need to add the `~children` prop to most of these to make it fully work

**Comment by tepafoo** [comment]
+1

Updating to JSX3 would be very useful!

Anything we can do to help with the PR getting merged into master and published to npm? I'm happy to help.

File: src/BsReactstrap__Alert.re [hunk 1]
```diff
@@ -14,36 +12,5 @@ external makeProps:
...
-        ~isOpen?,
-        ~toggle?,
-        ~tag?,
-        ~transition?,
-        (),
-      ),
-    children,
-  );
+  React.element =
+  "Alert";
```
Comment (rusty-key) [inline src/BsReactstrap__Alert.re]: Please add newlines at the end of the files too. 
If you are using vscode, you can add `"files.insertFinalNewline": true,` to your config

**Review by rusty-key — CHANGES_REQUESTED** [review]
Sorry for the late response, I somehow didn't receive notifications for this repo.

Yeah, to make it work, we need to deal with `children` first. 

Also, it would be nice to keep compatibility with jsx2 for some time. We can do it by adding a wrapper to each file and transform children to array: 

```
[@bs.module "reactstrap"] [@react.component]
external make:
  (~tag: 'a=?, ~fluid: bool=?, ~className: string=?, ~cssModule: 'b=?, ~children, unit) => React.element =
  "Container";

module Jsx2 = {
  let component = ReasonReact.statelessComponent(__MODULE__);

  let make = (~tag=?, ~fluid=?, ~className=?, ~cssModule=?, children) => {
    let children = React.array(children);
    ReasonReactCompat.wrapReactForReasonReact(
      make,
      makeProps(~tag?, ~fluid?, ~className?, ~cssModule?, ~children, ()),
      children,
    );
  };
}
```

What do you think? Can you do that?

**Comment by rusty-key** [comment]
Sorry, I hijacked your PR, hope it's ok. I did the following:
— ran refmt on everything
— added children (blindly, to all components, we can remove them later)
— added jsx2 wrappers (generated with this script: https://gist.github.com/rusty-key/20d228a416fc70d31a71056bb03c9853)

**Comment by liubko** [comment]
@rusty-key looks good to me

**Comment by rusty-key** [comment]
ok, let's try it

**Comment by Anahkiasen** [comment]
Nice, very cool!

**Comment by liubko** [comment]
published v1.0.2

**Comment by tepafoo** [comment]
Thank you!

File: src/BsReactstrap__Input.re [hunk 2]
```diff
@@ -27,44 +25,48 @@ external makeProps:
...
-    ) =>
-  ReasonReact.wrapJsForReason(
-    ~reactClass=input,
-    ~props=
+module Jsx2 = {
+  let component = ReasonReact.statelessComponent(__MODULE__);
+
+  let make =
+      (
+        ~_type=?,
```
Comment (nirvdrum) [inline src/BsReactstrap__Input.re:39]: FYI, this is a regression from the path taken as of #19.

**Pull request merged**